models/userModel.js

```diff
@@ -0,0 +1,36 @@
+const mongoose = require('mongoose');
+const bcrypt = require('bcrypt');
+const ObjectId = mongoose.Schema.Types.ObjectId;
+
+mongoose.models = {};
+mongoose.modelSchemas = {};
+
+const User = require('./userModel.js');
+
+const UserSchema = new mongoose.Schema({
+  username: {
+    type: String,
+    unique: true,
+    required: true,
+    index: true,
+    lowercase: true
+  },
+  password: {
+    type: String,
+    required: true
+  },
+  notes: [{ type: ObjectId, ref: '' }]
+});
+UserSchema.pre('save', function(next) {
+  bcrypt.hash(this.password, 11, (err, hash) => {
+    if (err) {
+      return next(err);
+    }
+    this.password = hash;
+    return next();
+  });
+});
+UserSchema.methods.isPassWordValid = function(passwordGuess) {
```

Comment:
Since we called this method `verifyPassword` in `routes.js` file, we should be consistent and rename this to match. Also, we're passing two arguments to this method, the plain text password `passwordGuess` and a callback function which we're passing to `bcrypt.compare`:

```JS
UserSchema.methods.verifyPassword = function(passwordGuess, callback) {
  bcrypt.compare(passwordGuess, this.password, callback);
};
```